humanlayer-wui/src/components/internal/SessionDetail/SessionDetail.tsx

```diff
@@ -304,6 +324,74 @@ function SessionDetail({ session, onClose }: SessionDetailProps) {
     [session.id, autoAcceptEdits], // Dependencies
   )
 
+  // Add hotkey to archive session ('e' key)
+  useHotkeys(
+    'e',
+    async () => {
+      // TODO(3): The timeout clearing logic (using confirmingArchiveTimeoutRef) is duplicated in multiple places.
+      // Consider refactoring this into a helper function to reduce repetition.
+
+      // Clear any existing timeout
+      if (confirmingArchiveTimeoutRef.current) {
+        clearTimeout(confirmingArchiveTimeoutRef.current)
+        confirmingArchiveTimeoutRef.current = null
+      }
+
+      // Check if session is active (requires confirmation)
+      const isActiveSession = [
+        SessionStatus.Starting,
+        SessionStatus.Running,
+        SessionStatus.Completing,
+        SessionStatus.WaitingInput,
+      ].includes(session.status)
+
+      const isArchiving = !session.archived
+
+      if (isActiveSession && !confirmingArchive) {
+        // First press - show warning
+        setConfirmingArchive(true)
+        toast.warning('Press e again to archive active session', {
```

Comment:
Wonder if this should be a `Dialog`? That would avoid some of the complexity around needing to keep track of whether the confirmation comes in and also makes sure that you don't have to worry about the toast to stick around. (Note also, it breaks accessibility a bit if we don't have the toast have some sort of clickable button for a mouse)